src/validate_constraints.c

```diff
@@ -2244,15 +2244,16 @@ merge_mswitches_with_and(struct jvst_cnode *mswlst)
 
 			assert(opts->carryopaque == merge_mcases);
 
-			dfa2 = fsm_clone_with_opts(n->u.mswitch.dfa, opts);
+			dfa2 = fsm_clone(n->u.mswitch.dfa);
 			if (dfa2 == NULL) {
 				perror("merging mswitch nodes (cloning DFA2)");
 				abort();
 			}
+			fsm_setoptions(dfa2, opts);
 
-			// NB: unlike fsm_intersect, fsm_intersect_bywalk is not destructive to
-			// its arguments
-			both = fsm_intersect_bywalk(dfa1,dfa2);
+			// NB: fsm_intersect using walk2 method is not
+			// destructive to its arguments
+			both = fsm_intersect(dfa1,dfa2);
```

Comment:
Uh oh. That's an API change that I missed. Could you open a libfsm ticket to mention that please? And I'll hunt down all instances using it, because those will now be leaking.

Reply:
Hmm... I probably should have mentioned it earlier.

It's worth asking... do you want it to be destructive?  fsm_intersect and fsm_subtract could still free their arguments.

Reply:
I think it ought to match `fsm_union()`, since it's the same family of functions. I realise `fsm_union` _could_ be written in terms of the walk2 API, but its current destructive O(1) implementation is quite attractive. So perhaps we ought to have those as destructive, or perhaps pass a flag to have them non-destructive?

Reply:
For the timebeing I just did this, to keep the same API semantics: https://github.com/katef/libfsm/commit/2a8d32da980a82d9aa5c6fb99c28ed4e80ea1557

Reply:
Sounds good.  Updated https://github.com/sfstewman/jvst/tree/pr-property-names although github isn't showing the commit. :(

Reply:
If/when we get a non-destructive way to do this, I can update the relevant jvst bits.